src/main/java/com/gregtechceu/gtceu/common/item/DataItemBehavior.java

```diff
@@ -81,27 +79,24 @@ public void appendHoverText(ItemStack stack, @Nullable Level level, List<Compone
     }
 
     @Override
-    public InteractionResult useOn(UseOnContext context) {
-        Level level = context.getLevel();
-        BlockPos pos = context.getClickedPos();
-        ItemStack stack = context.getItemInHand();
-        if (!level.isClientSide) {
-            MetaMachine machine = MetaMachine.getMachine(level, pos);
-            Pair<GTRecipeType, String> researchData = ResearchManager.readResearchId(stack);
-            if (machine instanceof MEPatternBufferPartMachine && researchData == null) {
-                stack.getOrCreateTag().putIntArray("pos", new int[] { pos.getX(), pos.getY(), pos.getZ() });
-            } else if (machine instanceof MEPatternBufferProxyPartMachine proxy) {
-                if (stack.hasTag()) {
-                    if (stack.getOrCreateTag().contains("pos", Tag.TAG_INT_ARRAY)) {
-                        int[] posArray = stack.getOrCreateTag().getIntArray("pos");
-                        BlockPos bufferPos = new BlockPos(posArray[0], posArray[1], posArray[2]);
-                        proxy.setBuffer(bufferPos);
+    public InteractionResult onItemUseFirst(ItemStack itemStack, UseOnContext context) {
+        if (context.getLevel().getBlockEntity(context.getClickedPos()) instanceof MetaMachineBlockEntity blockEntity) {
+            if (blockEntity.getOwner() != null && !blockEntity.getOwner().isPlayerInTeam(context.getPlayer())) {
```

Comment:
Use `!(MetaMachineBlock#canOpenOwnerMachine)` for this ownership condition so that we respect configs. It also checks null so you can replace the entire condition.
(also make that method static bc onion forgor)

Reply:
I'll make the `canOpenOwnerMachine` a static function and move it and `canBreakOwnerMachine` to the `IMachineOwner` interface, they logically belong there imo.
When someone looks for ownership related methods they shouldn't look in the `MetaMachineBlock`.